plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -343,13 +346,18 @@ def get_metadata(self, metadata_files):
         previous_skip_set = set(scratchpad.get(constants.PREVIOUS_SKIP_LIST, []))
         current_skip_set = set(self.config.get(constants.CONFIG_SKIP, []))
         self.current_revision = metadata_files.revision
-        # if the revision is positive, hasn't increased and the skip list doesn't include
-        # new types that weren't present on the last run...
-        # determine missing units
         missing_units = repo_controller.missing_unit_count(self.repo.repo_id)
+
+        last_sync = self.conduit.last_sync()
+        sync_due_to_unit_removal = False
+        if last_sync is not None:
+            last_sync = dateutils.parse_iso8601_datetime(last_sync)
+            sync_due_to_unit_removal = last_sync < self.repo.last_unit_removed
```

Comment:
i think if the was no any unit removal then last_unit_removed will be None. I guess you should this situation also in account when comparing to last_sync